src/api/routes/community.ts

```diff
@@ -407,7 +407,15 @@ export default (app: Router): void => {
      *             schema:
      *               $ref: '#/components/schemas/UbiCommunity'
      */
-    route.get('/address/:address', controller.findByContractAddress);
+    route.get(
+        '/address/:address',
+        (req, res, next) => {
+            (req as any).authTokenIsOptional = true;
+            next();
+        },
```

Comment:
should we create a middleware function? :thinking:

Reply:
good idea, I created a new middleware called optionalAuthentication, that set the authTokenIsOptional and call the authenticateToken middleware.
what do you think?

Reply:
Great. Exactly what I was thinking. Can you just replace the other routes that have `authTokenIsOptional` too, please?

Reply:
done